File: 63.js
function longestConsecutive(nums) {
    const set = new Set(nums);
    let longest = 0;
    for (const num of set) {
        if (!set.has(num - 1)) {
            let current = num, streak = 1;
            while (set.has(current + 1)) {
                current++;
                streak++;
            }
            longest = Math.max(longest, streak);
        }
    }
    return longest;
}

// Driver
console.log(longestConsecutive([100,4,200,1,3,2]));

File: 63.py
def longestConsecutive(nums):
    num_set = set(nums)
    longest = 0
    for num in num_set:
        if num - 1 not in num_set:
            current, streak = num, 1
            while current + 1 in num_set:
                current += 1
                streak += 1
            longest = max(longest, streak)
    return longest

# Driver
print(longestConsecutive([100,4,200,1,3,2]))
